src/kakarot/instructions/system_operations.cairo

```diff
@@ -812,13 +812,14 @@ namespace CreateHelper {
         // so we use popped_len to derive the way we should handle
         // the creation of evm addresses
         if (popped_len != 4) {
-            let (tx_info) = get_tx_info();
+            let (nonce) = IContractAccount.get_nonce(ctx.starknet_contract_address);
 
             let (evm_contract_address) = CreateHelper.get_create_address(
-                ctx.evm_contract_address, tx_info.nonce
+                ctx.evm_contract_address, nonce
             );
 
             let (contract_account_class_hash_) = contract_account_class_hash.read();
+            let (nonce) = IContractAccount.increment_nonce(ctx.starknet_contract_address);
```

Comment:
do we need this updated nonce value? if not, what's the point of storing it?

Reply:
I'll change `increment_nonce` to not return anything and just set